### PR TITLE
parsing duration to object

### PR DESCRIPTION
**Pull request by catcherholms007**
while parsing we can immediately receive next values: years, months, days, hours, minutes, seconds and flag isNegative. It helps us get this values without other dependencies (moment) or own computing methods

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,13 +22,22 @@ pxd('P2Y6M5DT12H35M30S')
...
+// => { years: 2, months: 6, days: 5, hours: 12, minutes: 35, seconds: 30, isNegative: 0 }
+
+pxd('P1Y2M3DT5H20M30.123S', true)
+// => { years: 1, months: 2, days: 3, hours: 5, minutes: 20, seconds: 30.123, isNegative: 0 }
 ```
 
 ## API
 
-### `parse-xsd-duration(xsdDuration: string): number | null`
+### `parse-xsd-duration(xsdDuration: string, toObject: boolean): number | null`
```
Comment (catcherholms007) [inline README.md:38]: added new parameter "toObject"

File: src/index.js [hunk 2]
```diff
@@ -7,6 +7,26 @@ const UNITS = {
   SECOND: 1
 }
 
+const YEAR_UNIT = 'years'
+const MONTH_UNIT = 'months'
+const DAY_UNIT = 'days'
+const HOUR_UNIT = 'hours'
+const MINUTE_UNIT = 'minutes'
+const SECOND_UNIT = 'seconds'
+const IS_NEGATIVE_UNIT = 'isNegative'
```
Comment (catcherholms007) [inline src/index.js:16]: added constants for fields names

File: src/index.js [hunk 3]
```diff
@@ -7,6 +7,26 @@ const UNITS = {
...
+  [MONTH_UNIT]: 0,
+  [DAY_UNIT]: 0
+}
+
+const emptyTime = {
+  [HOUR_UNIT]: 0,
+  [MINUTE_UNIT]: 0,
+  [SECOND_UNIT]: 0
+}
+
```
Comment (catcherholms007) [inline src/index.js:29]: If duration haven`t some values it returns the object with zero values

File: test/index.js [hunk 4]
```diff
@@ -2,6 +2,62 @@ import blons from 'big-list-of-naughty-strings'
...
+ * P20MT                                            no time items are present, so "T" must not be present
+ * P1YM5D                                           no value is specified for months, so "M" must not be present
+ * P15.5Y                                           only the seconds can be expressed as a decimal
+ * P1D2H                                            "T" must be present to separate days and hours
+ * 1Y2M                                             "P" must always be present
+ * P2M1Y                                            years must appear before months
+ * P                                                at least one number and designator are required
+ * PT15.S                                           at least one digit must follow the decimal point if it appears
+ */
+
```
Comment (catcherholms007) [inline test/index.js:60]: added duration description from official documentation with valid and invalid values

**Comment by mickdekkers** [comment]
This is a great addition, thank you for the PR! 😄

**Review by mickdekkers — APPROVED** [review]

**Pull request merged**